encoder.go

```diff
@@ -17,3 +17,5 @@ type Encoder interface {
 func defaultEncoderFunc(w io.Writer) Encoder {
 	return json.NewEncoder(w)
 }
+
+func (f EncoderFunc) apply(o *clientOptions) { o.newEncoder = f }
```

Comment:
We can just return 
optionFunc(func(o *clientOptions) {
		o. newEncoder = f
	}
})

Reply:
Not required!

Reply:
Also, the semantic won't allow